qiskit/circuit/quantumcircuit.py

```diff
@@ -1875,39 +1887,52 @@ def assign_parameters(self, param_dict, inplace=False):
         # replace in self or in a copy depending on the value of in_place
         bound_circuit = self if inplace else self.copy()
 
-        # unroll the parameter dictionary (needed if e.g. it contains a ParameterVector)
-        unrolled_param_dict = self._unroll_param_dict(param_dict)
-
-        # check that only existing parameters are in the parameter dictionary
-        if unrolled_param_dict.keys() > self._parameter_table.keys():
-            raise CircuitError('Cannot bind parameters ({}) not present in the circuit.'.format(
-                [str(p) for p in param_dict.keys() - self._parameter_table]))
+        if isinstance(params, dict):
+            # unroll the parameter dictionary (needed if e.g. it contains a ParameterVector)
+            unrolled_param_dict = self._unroll_param_dict(params)
 
-        # replace the parameters with a new Parameter ("substitute") or numeric value ("bind")
-        for parameter, value in unrolled_param_dict.items():
-            bound_circuit._assign_parameter(parameter, value)
+            # check that only existing parameters are in the parameter dictionary
+            if unrolled_param_dict.keys() > self._parameter_table.keys():
+                raise CircuitError('Cannot bind parameters ({}) not present in the circuit.'.format(
+                    [str(p) for p in params.keys() - self._parameter_table]))
 
+            # replace the parameters with a new Parameter ("substitute") or numeric value ("bind")
+            for parameter, value in unrolled_param_dict.items():
+                bound_circuit._assign_parameter(parameter, value)
+        else:
+            if len(params) != self.num_parameters:
+                raise ValueError('Mismatching number of values and parameters. For partial binding '
+                                 'please pass a dictionary of {parameter: value} pairs.')
+            for i, value in enumerate(params):
+                bound_circuit._assign_parameter(self._parameter_table[i], value)
         return None if inplace else bound_circuit
 
-    def bind_parameters(self, value_dict):
+    def bind_parameters(self, params):
```

Comment:
Does this need a `@deprecate_arguments`?

Reply:
Yes , added (also in `nlocal`)